### PR TITLE
Support for Kotlin DSL for Gradle for mn cli -build parameter. Implements #1019

### PR DESCRIPTION
**Pull request by jtonic**
Support for Kotlin DSL for Gradle for mn cli -build parameter. Implements #1019

### CODE REVIEW

**Comment by ilopmar** [comment]
I think we shouldn't merge this until we finish with all the features and have a clear picture of the "final version". If we merge this now we will have to maintain and modify it multiple times because of the new build option.

**Comment by jtonic** [comment]
> I think we shouldn't merge this until we finish with all the features and have a clear picture of the "final version". If we merge this now we will have to maintain and modify it multiple times because of the new build option.

I'm not very familiar with all that needed  to be done.
If there is anything I could help with, related to this new build type, just let me know.

**Comment by jameskleeh** [comment]
@jtonic We really appreciate this contribution! Picking up this API with no contributing documentation or any type of help is commendable. We are actively working on adding features to be compatible with the existing CLI so these files are rapidly evolving. The best thing to do would be to periodically check in and see what is new and missing from your build file and update the PR. I'm looking forward to merging this

**Comment by jtonic** [comment]
> @jtonic We really appreciate this contribution! Picking up this API with no contributing documentation or any type of help is commendable. We are actively working on adding features to be compatible with the existing CLI so these files are rapidly evolving. The best thing to do would be to periodically check in and see what is new and missing from your build file and update the PR. I'm looking forward to merging this

Thank you @jameskleeh for your kind words and recommendation.
I will keep the PR branch in sync with the future work and try to get acquainted with the evolution of the API and adapt my PR to the API evolution.

I have a question.
Next week I will do a company internal presentation about Micronaut and Kotlin. It is no problem for me to use the --build=gradle but I would really like to have Kotlin DSL for gradle as the build tool available for mn cli.
Can you give me some guidance about how can I build locally the mn cli so as to use the --build=gradle-kotlin... if it is possible with the new support in micronaut-starter?
Thanks

File: starter-core/src/main/java/io/micronaut/starter/feature/build/gradlekotlin/templates/buildGradleKts.rocker.raw [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+}
+    withType(JavaExec) {
+        classpath += configurations.developmentOnly
+        val jvmArgsList = mutableSetOf("-noverify", "-XX:TieredStopAtLevel=1", "-Dcom.sun.management.jmxremote") as LinkedHashSet
+@if (features.contains("springloaded")) {
+        jvmArgsList += listOf("-noverify", "-javaagent:agent/springloaded-$springLoadedVersion.jar")
+}
+@if (features.contains("jrebel")) {
+        if (project.hasProperty("rebelAgent")) {
+            jvmArgsList = listOf(rebelAgent)
```
Comment (jameskleeh) [inline starter-core/src/main/java/io/micronaut/starter/feature/build/gradlekotlin/templates/buildGradleKts.rocker.raw:242]: This should be `+=`

**Comment by jameskleeh** [comment]
@jtonic You can build the cli with `./gradlew starter-cli:assemble` and it will generate the `mn` command

**Comment by sgarfinkel** [comment]
Could this generator be enabled automatically when you use `mn -b gradle -l kotlin`? That's the default behavior of Spring Initializr.

**Comment by jameskleeh** [comment]
@sgarfinkel Yes anything is possible

**Comment by graemerocher** [comment]
Replaced by #491

**Pull request closed without merging**